lib/route/src/Obelisk/Route/Frontend.hs

```diff
@@ -337,6 +423,26 @@ runRouteViewT routeEncoder a = do
           setState = attachWith f ((,) <$> current historyState <*> current route) changeState
   return result
 
+-- | A link widget that, when clicked, sets the route to the provided route. In non-javascript
+-- contexts, this widget falls back to using @href@s to control navigation
+routeLink
+  :: forall t m a route.
+     ( DomBuilder t m
+     , RouteToUrl (R route) m
+     , SetRoute t (R route) m
+     )
+  => R route -- ^ Target route
+  -> m a -- ^ Child widget
+  -> m a
+routeLink r w = do
+  enc <- askRouteToUrl
+  let cfg = (def :: ElementConfig EventResult t (DomBuilderSpace m))
+        & elementConfig_eventSpec %~ addEventSpecFlags (Proxy :: Proxy (DomBuilderSpace m)) Click (\_ -> preventDefault)
+        & elementConfig_initialAttributes .~ "href" =: enc r
+  (e, a) <- element "a" cfg w
+  setRoute $ r <$ domEvent Click e
```

Comment:
You may need to explicitly `scrollTo(0)` here as well. Has this been tested on links very far down the page?

Reply:
Isn't this going to be a general issue with `setRoute` and not just this code?

Reply:
I'm presuming that not all `setRoute` use cases actually want to mimic a full "page load". But here it's quite obvious that you do want it.

Reply:
`setRouteWithOptions` ?

Reply:
I don't think we should _unconditionally_ `scrollTo` near `setRoute` or inside `routeLink`. It should be an option to be enabled or disabled. 

I have at least one use case where I would want my app to control where to scroll to. 

Here's [an example](https://github.com/srid/Taut/pull/24/files) where the app does automatic scrolling by processing a _top-level_ dynamic -- basically we either scroll to top (most routes) or to a specific element in the body (if a specific permalink route is active) -- and this would be messy to implement if obelisk was automagically scrolling for us.

Reply:
Definitely not `setRoute` but possibly `routeLink` since it is actually designed to work without JavaScript. I.e. the behavior with and without JS is supposed to be the same.